README.md

```diff
@@ -363,6 +364,89 @@ However, in the following scenario the default locale will be used, although it
+
+### Precaching
+
+The `precache` option can take the following options to control precaching behavior:
+
+| Property | Type | Optional | Description |
+| -------- | ---- | -------- | ----------- |
+| baseDir | `string` | Yes | The base directory to match `include` against. |
+| ignore | `string[]` | Yes | An array of glob pattern string matching files that should be ignored when generating the precache. Defaults to `[ 'node_modules/**/*' ]`. |
+| include | `string` or `string[]` | Yes | A glob pattern string or an array of glob pattern strings matching files that should be included in the precache. Defaults to all files in the build pipeline. |
+| index | `string` | Yes | The index filename that should be checked if a request fails for a URL ending in `/`. Defaults to `'index.html'`. |
+| maxCacheSize | `number` | Yes | The maximum size in bytes a file must not exceed to be added to the precache. Defaults to `2097152` (2 MB). |
+| strict | `boolean` | Yes | If `true`, then the build will fail if an `include` pattern matches a non-existent directory. Defaults to `true`. |
+| symlinks | `boolean` | Yes | Whether to follow symlinks when generating the precache. Defaults to `true`. |
+
+### Runtime Caching
+
+In addition to precaching, strategies can be provided for specific routes to determine whether and how they can be cached. This `routes` option is an array of objects with the following properties:
+
+| Property | Type | Optional | Description |
+| -------- | ---- | -------- | ----------- |
+| urlPattern | `string` | No | A pattern string (which will be converted a regular expression) that matches a specific route. |
+| strategy | `string` | No | The caching strategy (see below). |
+| options | `object` | Yes | An object of additional options, each detailed below. |
+| cacheName | `string` | Yes | The name of the cache to use for the route. Note that the `cachePrefix` is _not_ prepended to the cache name. Defaults to the main runtime cache (`${cachePrefix}-runtime-${domain}`). |
+| cacheableResponse | `object` | Yes | Uses HTTP status codes and or headers to determine whether a response can be cached. This object has two optional properties: `statuses` and `headers`. `statuses` is an array of HTTP status codes that should be considered valid for the cache. `headers` is an object of HTTP header and value pairs; at least one header must match for the response to be considered valid. Defaults to `{ statuses: [ 200 ] }` when the `strategy` is `'cacheFirst'`, and `{ statuses: [0, 200] }` when the `strategy` is either `networkFirst` or `staleWhileRevalidate`. |
+| expiration | `object` | Yes | Controls how the cache is invalidated. This object has two optional properties. `maxEntries` is the number of responses that can be cached at any given time. Once this max is exceeded, the oldest entry is removed. `maxAgeSeconds` is oldest a cached response can be in seconds before it is removed. |
```

Comment:
"`maxAgeSeconds` is oldest a cached response can be in seconds before it is removed" perhaps better as "`maxAgeSeconds` is the oldest a cached response can be in seconds before it gets removed"